### PR TITLE
Switch from Chef to Cinc

### PR DESCRIPTION
**Pull request by lukeseawalker**
Switch from Chef to Cinc

* switch from Chef 14.2.0 to Cinc 15.11.8
* allow custom install url to be specified when setting env variable CHEF_CUSTOM_INSTALL_URL

Signed-off-by: Luca Carrogu <carrogu@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: amis/packer_alinux.json [hunk 1]
```diff
@@ -174,13 +174,13 @@
       "type" : "shell",
       "inline" : [
         "region=\"{{user `region`}}\"",
-        "curl --retry 3 -L https://${region}-aws-parallelcluster.s3.${region}.amazonaws.com$([ \"${region}\" != \"${region#cn-*}\" ] && echo \".cn\")/archives/chef/chef-install.sh | sudo bash -s -- -v {{user `chef_version`}}"
+        "curl --retry 3 -L https://${region}-aws-parallelcluster.s3.${region}.amazonaws.com$([ \"${region}\" != \"${region#cn-*}\" ] && echo \".cn\")/archives/cinc/cinc-install.sh | sudo bash -s -- -v {{user `chef_version`}}"
```
Comment (tilne) [inline amis/packer_alinux.json:177]: could we add a user variable for the bucket to download from and default to `${region}-aws-parallelcluster`?

Comment (lukeseawalker) [reply amis/packer_alinux.json]: I have added a variable that can be used to specify the whole URL, please check

**Comment by tilne** [comment]
One change to make that I found this morning:

[build_ami.sh](https://github.com/aws/aws-parallelcluster-cookbook/blob/develop/amis/build_ami.sh#L38) mentions chef DK. It would be good to change that to mention cinc instead.

That said, the conditional is checking for `berks`, and the [createami documentation](https://docs.aws.amazon.com/parallelcluster/latest/ug/pcluster.createami.html) suggests installing it via `gem`. Do we need to recommend installing cincdk or can we just say install berks by itself?

**Comment by lukeseawalker** [comment]
> One change to make that I found this morning:

@tilne thanks, fixed and aligned with doc

**Pull request merged**